### ISSUE TITLE
Re-write N3UG web site

### ISSUE THREAD

**Issue by stevefabian** [n3ug/N3UG.Projects#1]
start a project where we re-write the N3UG web site using newer technologies, a more flexible architecture and possibly in such a way that it would be easily re-usable for any other User Group or Club to manage members and share information, meeting schedules and resources.

**Comment by DavidTaylor1** [comment]
Great idea Steven.  Since this is my first time at this, where do we bring?

**Comment by cicorias** [comment]
Have you considered static site generators? One for .net  https://www.hanselman.com/blog/ExploringWyamANETStaticSiteContentGenerator.aspx

Whole bunch for JavaScript and Gatsby is hot

For UI, React, Angular, Vue?

**Comment by DavidTaylor1** [comment]
Shawn... thanks for this info.  Static content generation CMS seems like a good alternative to writing code.  I am peculiarly interested in Wyman...looks good

**Comment by DavidTaylor1** [comment]
I would still like to hear more about what we are looking to achieve before we decide on the technology and implementation strategy though

**Comment by n3ug** [comment]
the current N3UG web site, is a DotNetNuke site with a custom skin.  I thought it might be worth a re-write using modern technologies that we've all seen presented at N3UG over the past year.  There are so many choices in platforms, frameworks, etc.  it might be nice to see if we started with a clean slate and wanted to build a web site, what would the architecture look like and what development technologies would we pick for each component.  .Net Core for the server layer? REST based API services, oAuth, front end UI choices like Angular vs React, vs Vue vs anything else.  So at this point it's a blank canvas :) open for discussion.  

I'm not saying this is the first project we take on as a User Group community, I threw this out there to start the conversation. I'm hoping the members will throw out some other ideas as well for open source projects, and it may even be something as small as a Visual Studio extension.  Then if we get some interest behind one or more projects, we can start a repository for each one and begin identifying features and making some tech and tool decisions.

**Comment by DavidTaylor1** [comment]
Thanks for the clarification.  One thing of interest to me is IoT particular with Windows 10 Core and Rasberry Pi 3.   Don't know if this is something the group also find to be off interest as well

**Comment by stevefabian** [comment]
Great idea, I would suggest creating a new issue here and let's see if anyone is interested

**Comment by DavidTaylor1** [comment]
Thanks Stephen.  I just created a Windows 10 Core IoT issue.  Hopefully the topic will stir some interest among the group.

**Comment by fruscil-jr** [comment]
Would there be any interest in developing a "Xamarin or mobile version" of the N3UG website?
I like the IoT idea as well.

**Comment by stevefabian** [comment]
@fruscil-jr Absolutely! These days I think when designing and build any software application, you have to think mobile, so re-writing N3UG should absolutely respect mobile devices

**Comment by stevefabian** [comment]
At yesterday's meeting it was suggested that we use Angular for the client side. That's certainly a fine choice. Angular, React and Vue seem to be the popular choices for client side frameworks, and any of them would be a good choice. When it comes time to select technologies we'll have a brief poll and see what the majority of contributors want to use.  Thanks for the suggestion.
